web3/contract.py

```diff
@@ -1132,6 +1133,46 @@ def _parse_logs(self, txn_receipt):
                 continue
             yield decoded_log
 
+    @combomethod
+    def createFilter(
+            self, *,  # PEP 3102
+            argument_filters=dict(),
```

Comment:
This should get fixed.  Mutable values as defaults for function arguments is an antipattern:

http://docs.python-guide.org/en/latest/writing/gotchas/

This specific case, I don't think it causes any problems, but still something we should avoid.

Reply:
createEvent does not yield the same result as eventFilter...For example, the below gives no results with createFilter and some results with eventFilter....
```
myfilter_new=contract_instance.events.Deposit.createFilter(fromBlock=2058615)
myfilter_old=contract_instance.eventFilter('Deposit',{'fromBlock':2058615,'toBlock':'latest'})
print("     filterDeposit (OLD METHOD)--",myfilter_old.get_all_entries())
print("     filterDeposit (NEW METHOD)--",myfilter_new.get_all_entries())
```